edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Kubernetes.Test/KubernetesExperimentalPodParametersTest.cs

```diff
@@ -168,5 +169,92 @@ public void ParsesSomeResourcesExperimentalOptions()
             parameters.Resources.ForEach(resources => Assert.Equal(new Dictionary<string, ResourceQuantity> { ["memory"] = new ResourceQuantity("128Mi") }, resources.Limits));
             parameters.Resources.ForEach(resources => Assert.Equal(new Dictionary<string, ResourceQuantity> { ["cpu"] = new ResourceQuantity("250m") }, resources.Requests));
         }
+
+        [Fact]
+        public void ParsesNoneVolumesExperimentalOptions()
+        {
+            var experimental = new Dictionary<string, JToken>
+            {
+                ["k8s-experimental"] = JToken.Parse("{ volumes: null }")
+            };
+
+            var parameters = KubernetesExperimentalCreatePodParameters.Parse(experimental).OrDefault();
+
+            Assert.False(parameters.Volumes.HasValue);
+        }
+
+        [Fact]
+        public void ParsesEmptyVolumesExperimentalOptions()
```

Comment:
This is just parser and it doesn't make any validations at all. If we want to make such test it should be inside mapper tests, because mapper has knowledge about all volumes/mounts. 

Regarding having a volumeMount section is empty is OK. Because it basically means the user mounted some folder to a pod and left it alone. The volume mount is the one that links this folder into a container. In other words, it is a legitimate situation. k8s api will not respond any errors.
Otherwise, when volumeMount points to a volume that doesn't exist - is a clear error. But if we are going to validate this kind of errors we will end up re-implementing a bunch of checks that are part of k8s api now. So we can delegate validation to k8s itself. If there is an error, an agent should report it to both iothub and crd status.